crates/router/src/connector/rapyd/transformers.rs

```diff
@@ -442,44 +447,46 @@ impl TryFrom<types::PaymentsCaptureResponseRouterData<RapydPaymentsResponse>>
     fn try_from(
         item: types::PaymentsCaptureResponseRouterData<RapydPaymentsResponse>,
     ) -> Result<Self, Self::Error> {
-        let (status, response) = match item.response.status.status.as_str() {
-            "SUCCESS" => match item.response.data {
-                Some(data) => (
-                    enums::AttemptStatus::foreign_from((data.status, data.next_action)),
-                    Ok(types::PaymentsResponseData::TransactionResponse {
-                        resource_id: types::ResponseId::ConnectorTransactionId(data.id), //transaction_id is also the field but this id is used to initiate a refund
-                        redirection_data: None,
-                        redirect: false,
-                        mandate_reference: None,
-                        connector_metadata: None,
-                    }),
-                ),
-                None => (
-                    enums::AttemptStatus::Failure,
-                    Err(types::ErrorResponse {
-                        code: item.response.status.error_code,
-                        message: item.response.status.status,
-                        reason: item.response.status.message,
-                        status_code: item.http_code,
-                    }),
-                ),
-            },
-            "ERROR" => (
+        let (status, response) = match &item.response.data {
```

Comment:
`RapydPaymentsResponse` is same for mostly of the api's the handling also not differs, write a common function and use it.

Reply:
sure

Reply:
sure